src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,373 @@
+            if (scenarioTestsPackage == null)
+            {
+                Assert.Fail("Scenario tests package not found");
+            }
+
+            ZipFile.ExtractToDirectory(scenarioTestsPackage, Path.Combine(_contextDir, "scenario-tests"));
+            _sharedContextInitialized = true;
+        }
+    }
+
+    private void InsertLocalPackagesPathToNuGetConfig(string nuGetConfig, string localPackagesPath)
+    {
+        XDocument doc = XDocument.Load(nuGetConfig);
+        if (doc.Root != null)
+        {
+            XElement? packageSourcesElement = doc.Root.Element("packageSources");
+            if (packageSourcesElement != null)
+            {
+                XElement? clearElement = packageSourcesElement.Element("clear");
+                if (clearElement != null)
+                {
+                    XElement newAddElement = new XElement("add",
+                    new XAttribute("key", "local-packages"),
+                        new XAttribute("value", localPackagesPath));
+
+                    clearElement.AddAfterSelf(newAddElement);
+                }
+            }
+
+            doc.Save(nuGetConfig);
+        }
+    }
+
+    private void DistroTest(string baseImage, PackageType packageType)
+    {
+        // Order of installation is important as we do not want to use "--nodeps"
+        // We install in correct order, so package dependencies are present.
+
+        // Prepare the package list in correct install order
+        List<string> packageList =
+        [
+            // Deps package should be installed first
+            Path.GetFileName(GetMatchingDepsPackage(baseImage, packageType))
+        ];
+
+        // Add all other packages in correct install order
+        AddPackage(packageList, "dotnet-host-", packageType);
+        AddPackage(packageList, "dotnet-hostfxr-", packageType);
+        AddPackage(packageList, "dotnet-runtime-", packageType);
+        AddPackage(packageList, "dotnet-targeting-pack-", packageType);
+        AddPackage(packageList, "aspnetcore-runtime-", packageType);
+        AddPackage(packageList, "aspnetcore-targeting-pack-", packageType);
+        AddPackage(packageList, "dotnet-apphost-pack-", packageType);
+        if (Config.Architecture == "x64")
+        {
+            // netstandard package exists for x64 only
+            AddPackage(packageList, "netstandard-targeting-pack-", packageType);
+        }
+        AddPackage(packageList, "dotnet-sdk-", packageType);
+
+        string dockerfile = GenerateDockerfile(packageList, baseImage, packageType);
+
+        string tag = $"test-{Path.GetRandomFileName()}";
+        string output = "";
+        bool buildCompleted = false;
+
+        try
+        {
+            // Build docker image and run the tests
+            _dockerHelper.Build(tag, dockerfile: dockerfile, contextDir: _contextDir);
+            buildCompleted = true;
+            output = _dockerHelper.Run(tag, tag);
+
```

Comment:
Would it make sense to capture the scenario-tests xml log file itself.  Would this help in diagnosing scenario-test failures?

Reply:
I don't think this is needed, as we dump the whole output, which would contain the complete failure log.